integration/spark2/src/main/scala/org/apache/spark/sql/CarbonEnv.scala

```diff
@@ -135,17 +135,25 @@ class CarbonEnv {
       dataMapSchema =>
         if (null != dataMapSchema.getRelationIdentifier &&
             !dataMapSchema.isIndexDataMap) {
-          if (!sparkSession.sessionState
-            .catalog
-            .tableExists(TableIdentifier(dataMapSchema.getRelationIdentifier.getTableName,
-              Some(dataMapSchema.getRelationIdentifier.getDatabaseName)))) {
+          val isTableExists = try {
+            sparkSession.sessionState
+              .catalog
+              .tableExists(TableIdentifier(dataMapSchema.getRelationIdentifier.getTableName,
+                Some(dataMapSchema.getRelationIdentifier.getDatabaseName)))
+          } catch {
+            // we need to take care of cleanup when the table does not exists, if table exists and
+            // some other user tries to access the table, it might fail, that time no need to handle
+            case ex: Exception =>
```

Comment:
Should catch NoSuchTableException?

Reply:
no, basically this will happen only when the table not exists in hive metastore and schema is present in carbon store. So we just need to proceed only when the tableexists is false, else for anything its fine.